tests/v3/test_store/test_stateful_store.py

```diff
@@ -0,0 +1,209 @@
+# Stateful tests for arbitrary Zarr stores.
+
+import asyncio
+
+import hypothesis.strategies as st
+from hypothesis import assume, note
+from hypothesis.stateful import (
+    RuleBasedStateMachine,
+    invariant,
+    precondition,
+    rule,
+)
+
+from zarr.core.buffer import Buffer, BufferPrototype, default_buffer_prototype
+from zarr.store import MemoryStore
+from zarr.testing.strategies import key_ranges, paths
+
+
+class SyncStoreWrapper:
```

Comment:
You might consider changing this to: 
```suggestion
class SyncStoreWrapper(zarr.core.sync.SyncMixin):
```

Which will give you access to the `_sync` and `_sync_iter` methods out of the box.

Reply:
Should we use `_sync` or `zarr.core.sync.sync`